validation/unique.go

```diff
@@ -144,6 +148,68 @@ func (v *ExistsArrayValidator[T]) buildQuery(values []T, condition bool) *gorm.D
 	return db.Raw(sql, params...)
 }
 
+// Build Query for Clickhouse syntax
+func (v *ExistsArrayValidator[T]) buildClickhouseQuery(values []T, condition bool) *gorm.DB {
+	questionMarks := []string{}
+	params := []any{}
+
+	paramZeroType := ""
+	paramOneType := ""
+
+	for i, val := range values {
+		questionMarks = append(questionMarks, "?")
+		var transformedValue any = val
+		if v.Transform != nil {
+			transformedValue = v.Transform(val)
+		}
+		params = append(params, gorm.Expr(
+			"(?,?)",
+			transformedValue, i))
+		if paramZeroType == "" {
+			paramZeroType = clickhouseTypes[reflect.TypeOf(transformedValue)]
```

Comment:
This can be done once outside the loop. It would be cleaner in my opinion:
```go
var zeroVal T
paramType, ok := clickhouseTypes[reflect.TypeOf(zeroVal)
```
You should check if the value is supported, otherwise [add an error](https://pkg.go.dev/goyave.dev/goyave/v5@v5.1.1/validation#Context.AddError) to the validation context if no `Transform` function is provided.

`transformedValue` is a `gorm.Expr` so your reflect call here will never work if you provide a transform function.

Reply:
So if the value is not supported within the `clickhouseTypes` map, but there is a Transform function provided, will we be able to determine what `paramType` should be?

Reply:
Transform is an option for special cases. For example for Postgres it allows to do the following:
```go
UniqueArray("table", "column", func(val int64) clause.Expr {
	return gorm.Expr("?::bigint", val)
})
```
```go
UniqueArray("table", "column", func(val string) clause.Expr {
	return gorm.Expr("LOWER(?)", val)
})
```

So in the case of clickhouse, this would allow developers to cast raw values into special types that we don't support directly.